pkgs/by-name/pw/pwalarmctl/package.nix

```diff
@@ -0,0 +1,43 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+, pkg-config
+, alsa-lib
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "pwalarmctl";
+  version = "0.1.0";
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ alsa-lib ];
+
+  src = fetchFromGitHub {
+    owner = "amyipdev";
+    repo = "pwalarmd";
+    rev = "v${version}";
+    hash = "sha256-xoC1PtDQjkvoWb9x8A43ITo6xyYOv9hxH2pxiZBBvKI=";
+  };
+
+  configurePhase = ''
+    cargo check
+  '';
```

Comment:
Looks like this should be in `checkPhase`?

Reply:
@wegank ~~it appears checkPhase is a post-build thing for tests? if so then that is not what this is for.~~ *okay so it does run before installPhase but not sure if it's universal, and it very much obscures the point of what this is for, which if there was a generalized preInstallPhase this would communicate that better*. this is for building dependencies in the outside directory - least intensive workaround i can find to make it work as suggested by several forum articles

also #33599 -> https://github.com/NixOS/rfcs/pull/95 and https://discourse.nixos.org/t/nix-develop-and-checkphase/25707

Reply:
Oh, okay. Does `preBuild` work here instead of `configurePhase`? Otherwise, you need to add `runHook preConfigure` and `runHook postConfigure` in it.

Reply:
@wegank  I kept trying to look for something like `preBuild`. I'll test that out and get back.

Also, I never had to add `runHook preConfigure` - it just automatically runs `preConfigure` for me.

Reply:
That's the case when you don't override `configurePhase`, but I don't think `preConfigure = "exit 1";` will stop the build in the current derivation.

Reply:
So, does something need to be changed?